challenges/cloud/challenge.md

```diff
@@ -0,0 +1,379 @@
+document should include a section describing how this solution could be deployed
+to an AWS EKS cluster.
+
+### Server
+
+* HTTP API to retrieve the replica count of the Kubernetes Deployment
+* HTTP API to set the replica count of a the Kubernetes Deployment
+* HTTP API to get the list of available Deployments in the Kubernetes cluster
+* HTTP health check verifying Kubernetes connectivity
+* Extend the API to support a difference between current and desired state
+* Extend the Server to support the ability to show the differences between current and desired state
+* Secure connections with mTLS
+
+### Automation
+
+* Ability to build the server in a Docker container
+* Ability to run the server
+* Ability to execute tests against a locally configured Kubernetes config
+* GitHub Actions for every commit that verifies code quality, ensures the software builds, and all tests pass
+* Build releases of the Server
+* Produce production quality Docker images
+
+### Deployment
+
+* Create a helm chart for the service
+* Deploy releases of the API Server to a Kubernetes cluster (and possible dependencies)
+* Upgrade releases of the API Server to a Kubernetes cluster (and possible dependencies)
+
+
+## Level 5
+
+### Design
+
+In addition to including details about the requirements below, the design
+document should include a section describing how this solution could be deployed
+to an AWS EKS cluster. Please include security considerations as well.
+
+### Server
+
+* HTTP API to retrieve the replica count of the Kubernetes Deployment
+* HTTP API to set the replica count of a the Kubernetes Deployment
+* HTTP API to get the list of available Deployments in the Kubernetes cluster
+* HTTP health check verifying Kubernetes connectivity
+* Extend the API to support a difference between current and desired state
+* Extend the Server to support the ability to show the differences between current and desired state
+* Secure connections with mTLS
+* Replace the HTTP API with gRPC
+* Extend the Server to support reconciling cluster state (i.e. an external actor changed the replica count manually)
+
+### Automation
+
+* Ability to build the server in a Docker container
+* Ability to run the server
+* Ability to execute tests against a locally configured Kubernetes config
+* GitHub Actions for every commit that verifies code quality, ensures the software builds, and all tests pass
+* Build releases of the Server
+* Produce production quality Docker images
+* Deploy and manage a local Kubernetes Cluster
+* Deploy and upgrade to the local Kubernetes Cluster with no service interruption
+* GitHub Action driven workflows for Deployment
+
+### Deployment
+
+* Create a helm chart for the service
+* Deploy releases of the API Server to a Kubernetes cluster (and possible dependencies)
+* Upgrade releases of the API Server to a Kubernetes cluster (and possible dependencies)
+* Include a production Deployment of this service: Service, Ingress, TLC, RBAC
```

Comment:
Thinking this should be w/L4 as well. We may also want to be more specific about each of the items. Not sure ingress should be included, and RBAC could be interpreted in many different ways.

Reply:
How about:

Include a production Deployment of this service, including (not limited to) these objects:
 - Deployment
 - Role
 - RoleBinding
 - ServiceAccount
 - Service

We can avoid the Ingress by just requiring the Service